frontend/src/features/plots/plotsSlice.ts

```diff
@@ -12,7 +12,7 @@ const slice = createSlice({
   reducers: {
     setCurrentPlot: (state, action) => {
       const id = action.payload
-      if (state.data && state.data.hasOwnProperty(id)) {
+      if (state.data && id in state.data) {
```

Comment:
I write to mark all these so if you think that the old version is the correct one you can revert all.

Reply:
Similar: we assume that this is a pure Redux object so it doesn't need to be restrictive.